### ISSUE TITLE
meson.build:67:0: ERROR: C library 'm' not found

### ISSUE THREAD

**Issue by pletcher** [swaywm/wlroots#1483]
Gist of meson-log.txt: https://gist.github.com/pletcher/4891cfee4a1083531ea7d7e9e8a36926

```
$ meson --version
> 0.49.0
```
OS: Arch
Kernel 4.20.3-arch1-1-ARCH

It looks like `cc.find_library('m')` is failing when installing from the AUR.

I installed wlroots and sway from the AUR a couple of weeks ago, but haven't been able to update to the latest releases because of this issue.

Any pointers? Is it safe to just overwrite that line to look for `'libm'` and build from source, or would that cause a mess with my current installation?

Thanks!

**Comment by ascent12** [comment]
There is some really weird output in that meson log.
Are you using a compiler that doens't come from the official Arch packages or a libc which isn't glibc?

**Comment by pletcher** [comment]
Nope, just straight glibc, unless I mistakenly screwed something up somewhere along the way.

**Comment by ascent12** [comment]
Looking into it slightly more, clang doesn't support `-fvar-tracking-assingments`, which makes the command to check for libm fail, rather than not actually being able to find it. It just happened that this check was the first required one.
It's being inserted from the CFLAGS environment variable, so you need to remove it from there.

**Comment by pletcher** [comment]
:+1: Thanks, all -- and yeah, it might be on meson's end. I'll try adjusting CFLAGS and check meson's open issues too.
